autofit/optimize/non_linear/emcee.py

```diff
@@ -206,23 +289,71 @@ def backend(self):
 
     @property
     def pdf(self):
+        """An interface to *GetDist* which can be used for analysing and visualizing the non-linear search chains.
+
+        *GetDist* can only be used when chains are converged enough to provide a smooth PDF and this convergence is
+        checked using the *pdf_converged* bool before *GetDist* is called.
+
+        https://github.com/cmbant/getdist
+        https://getdist.readthedocs.io/en/latest/
+
+        For *emcee*, chains are passed to *GetDist* via the hdt backend. *GetDist* currently does not provide accurate
+        model sampling.
+        """
         import getdist
 
         return getdist.mcsamples.MCSamples(samples=self.samples_after_burn_in)
 
     @property
     def pdf_converged(self):
+        """ To analyse and visualize chains using *GetDist*, the analysis must be sufficiently converged to produce
+        smooth enough PDF for analysis. This property checks whether the non-linear search's chains are sufficiently
+        converged for *GetDist* use.
+
+        Emcee chains can be analysed by GetDist irrespective of how long the sampler has run, albeit low run times
+        will likely produce inaccurate results."""
         return True
 
+    @property
+    def total_samples(self):
+        """The total number of samples performed by the non-linear search.
+
+        For Emcee, this includes all accepted and rejected proposed steps and is loaded from the results backend.
+        """
+        return len(self.backend.get_chain(flat=True))
+
+    @property
+    def total_walkers(self):
+        """The total number of walkers used by this *Emcee* non-linear search.
+        """
+        return len(self.backend.get_chain()[0, :, 0])
+
+    @property
+    def total_steps(self):
```

Comment:
Would be nice to put type annotations on functions

-> int

Reply:
Done.